com.unity.mobile.notifications/Runtime/iOS/Plugins/UnityNotificationData.m

```diff
@@ -213,7 +209,7 @@ void freeiOSNotificationData(iOSNotificationData* notificationData)
 
 void _ReadNSDictionary(void* csDict, void* nsDict, void (*callback)(void* csDcit, const char*, const char*))
 {
-    NSDictionary* dict = (__bridge NSDictionary*)nsDict;
+    NSDictionary* dict = (__bridge_transfer NSDictionary*)nsDict;
```

Comment:
I'm trying to understand the problem, and not quite sure why freeing in _ReadNSDictionary solved the problem. Could you please explain a little bit more?

Reply:
The issue was that entire notification data struct was freed too early, before the dictionary is read, resulting in crash when reading the dictionary afterwards (in some cases, since somtimes the dictionary is kept alive by objc side).
It's a bit messy and I intend to do a bigger refactoring later to untangle this thing. Just that for now the priority is to ship a fix to users, since push notifications have a major regression now.

Reply:
Agreed it's a bit messy. I originally thought that the lastReceivedNotification(which is UNNotification) on UnityNotificationManager would hold the reference to the dictionary, which could keep it away from being released.

Anyway, improving the lifetime control would be perfect as you mentioned.